src/main/java/org/spongepowered/api/text/translation/dictionary/CachingTranslationDictionary.java

```diff
@@ -0,0 +1,36 @@
+/*
+ * This file is part of SpongeAPI, licensed under the MIT License (MIT).
+ *
+ * Copyright (c) SpongePowered <https://www.spongepowered.org>
+ * Copyright (c) contributors
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+package org.spongepowered.api.text.translation.dictionary;
+
+/**
+ * A {@link TranslationDictionary} that can cache values.
+ */
+public interface CachingTranslationDictionary extends TranslationDictionary {
```

Comment:
Do remember that this is a multi-source translation dictionary, and not all resource bundles use `ResourceBundle#getBundle`, which means we need our own internal cache. Not using a cache would result in the translation dictionary loading the properties file from the filesystem and creating a `PropertyResourceBundle` each time, which is bad. With `CachingTranslationDictionary`, the cache can be cleared if any new sources are added (plugin adds a command or something to reload it's configuration, it would want to clear the cache).